src/preprocessing/passes/bv_to_int.cpp

```diff
@@ -0,0 +1,981 @@
+/*********************                                                        */
+/*! \file bv_to_int.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Yoni Zohar and Ahmed Irfan
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief The BVToInt preprocessing pass
+ **
+ ** Converts bit-vector operations into integer operations.
+ **
+ **/
+
+#include "preprocessing/passes/bv_to_int.h"
+
+#include <cmath>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "expr/node.h"
+#include "theory/bv/theory_bv_rewrite_rules_operator_elimination.h"
+#include "theory/bv/theory_bv_rewrite_rules_simplification.h"
+#include "theory/rewriter.h"
+
+namespace CVC4 {
+namespace preprocessing {
+namespace passes {
+
+using namespace CVC4::theory;
+using namespace CVC4::theory::bv;
+
+Rational intpow2(uint64_t b)
+{
+  Integer one = Integer(1);
+  Integer two = Integer(2);
+  Integer p = two.pow(b);
+  Rational r = Rational(p, one);
+  return r;
```

Comment:
You could make a one liner out of this:
`return Rational(Integer(2).pow(b), Integer(1));`
Less verbose, IMO a bit faster to comprehend, but that's of course personal preference.

Reply:
done.